libqtile/bar.py

```diff
@@ -249,7 +249,15 @@ def _configure(self, qtile, screen):
             self.crashed_widgets = []
             for idx, i in enumerate(self.widgets):
                 if i.configured:
-                    i = i.create_mirror()
+                    # To prevent rendering issues, we only create a mirror if
+                    # the bar backgrounds are identical
+                    if i.bar.background == self.background:
```

Comment:
This limits creation of `Mirror` instances to situations where the bar colour is identical, otherwise the widget is cloned.

Reply:
This is a bit of a simplification. Arguably, we could mirror if the widget has set it's own `background` colour as this suggests the user wants to override the bar background. However, I think we'd have issues with this if the mirror has a (semi-)transparent background on a solid (24 bit bar). The conditional testing starts to get a bit complicated at that point!

Reply:
> However, I think we'd have issues with this if the mirror has a (semi-)transparent background on a solid (24 bit bar).

How can it happen if the widget background overrides the bar background?

Reply:
I was worried that a mirror could have a 32 bit drawer but be drawn on a 24 bit bar, but maybe that's not actually possible. Let me test it.

Reply:
I hadn't appreciate the mirror creates a new `drawer` when it's configured. That `drawer` will be linked to the parent bar so the colour depth issue will be handled at that point.